kubelink/pkg/service/helmApplicationService/utils.go

```diff
@@ -33,3 +47,156 @@ const (
+	if err != nil {
+		return fmt.Errorf("error reading Chart.lock file at %s: %w", lockFilePath, err)
+	}
+	helmRelease.Chart.Lock = &chart2.Lock{}
+	if err := yaml.Unmarshal(lockFileData, helmRelease.Chart.Lock); err != nil {
+		return fmt.Errorf("error unmarshalling Chart.lock file at %s: %w", lockFilePath, err)
+	}
+	return nil
+}
+
+// ProcessTGZFiles locates and processes .tgz files in the charts directory.
+func ProcessTGZFiles(chartsDir string, helmRelease *release.Release) error {
+	files, err := os.ReadDir(chartsDir)
+	if err != nil {
+		return fmt.Errorf("error reading charts directory in dir %s :%w", chartsDir, err)
+
+	}
+	for _, file := range files {
+		if strings.HasSuffix(file.Name(), ".tgz") {
+			tgzPath := filepath.Join(chartsDir, file.Name())
+
+			if err := expandAndSetDependency(tgzPath, helmRelease); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// expandAndSetDependency expands a .tgz file and sets it as a dependency.
+func expandAndSetDependency(tgzPath string, helmRelease *release.Release) error {
+	// Create a temporary directory for expanding the chart
+	tempDir, err := os.MkdirTemp("", "helmDependencyChart*")
+	if err != nil {
+		return fmt.Errorf("error creating temporary directory: %w", err)
+	}
+	defer func() {
+		// Clean up the temporary directory
+		if removeErr := os.RemoveAll(tempDir); removeErr != nil {
+			fmt.Sprintf("error in removing temporary directory %s: %s", tempDir, removeErr.Error())
+		}
+	}()
+
+	// Open the .tgz file
+	tgzFile, err := os.Open(tgzPath)
+	if err != nil {
+		return fmt.Errorf("error opening tgz file of tgzPath %s: %w", tgzPath, err)
+	}
+	defer func(tgzFile *os.File) {
+		err := tgzFile.Close()
+		if err != nil {
+			fmt.Errorf("error in closing tgz file of tgzPath %s: %w", tgzPath, err)
+		}
+	}(tgzFile)
+
+	// Expand the .tgz file into the temporary directory
+	if err := chartutil.Expand(tempDir, tgzFile); err != nil {
+		return fmt.Errorf("error expanding tgz file having filePath %s : %w", tgzPath, err)
+	}
+
+	// Dynamically find the expanded chart directory
+	entries, err := os.ReadDir(tempDir)
+	if err != nil {
+		return fmt.Errorf("error reading contents of temporary directory %s: %w", tempDir, err)
+	}
+
+	var expandedChartDir string
+	for _, entry := range entries {
+		if entry.IsDir() {
+			expandedChartDir = filepath.Join(tempDir, entry.Name())
+			break
+		}
+	}
+
+	if expandedChartDir == "" {
+		err := fmt.Errorf("no expanded chart directory found in %s", tempDir)
+		return fmt.Errorf("error locating expanded chart directory found in %s: %w", tempDir, err)
+	}
+
+	// Load the expanded chart
+	expandedChart, err := loader.LoadDir(expandedChartDir)
+	if err != nil {
+		return fmt.Errorf("error loading expanded chart : %w", err)
+	}
+	helmRelease.Chart.SetDependencies(expandedChart)
+	return nil
+}
+
+func UpdateChartDependencies(providers getter.Providers, helmRelease *release.Release, registry *registry.Client) error {
```

Comment:
also seems like a business logic. should be a part of our interface/ implementation